gwcs/spectroscopy.py

```diff
@@ -192,3 +196,223 @@ def input_units(self):
             return {'wavelength': 1 / self.groove_density.unit,
                     'alpha_in': u.Unit(1),
                     'beta_in': u.Unit(1)}
+
+
+class Snell3D(Model):
+    """
+    Snell model in 3D form.
+
+    Inputs are direction cosines.
+
```

Comment:
This seems to presume that this is in a vacuum. Should the term be relative index of refraction? (e.g., ratio of indices of both materials?)

Reply:
I'm not an expert but I guess this is related to the comments below. `SellmeierZemax` returns relative index which can be passed in this equation. `SellmeierGlass` is the general law for glass.
According to the reference it is the refractive index.